### PR TITLE
[Bugfix] Fix ConcurrentModificationException of concurrently accessing childCounterMap

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/891

### CODE REVIEW

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java [hunk 1]
```diff
@@ -123,9 +122,11 @@ public Counter addCounter(String name, TUnit type, String parentName) {
             Counter newCounter = new Counter(type, 0);
             this.counterMap.put(name, Pair.create(newCounter, parentName));
 
-            TreeSet<String> childNames = childCounterMap.getOrDefault(parentName, new TreeSet<>());
+            if (!childCounterMap.containsKey(parentName)) {
+                childCounterMap.putIfAbsent(parentName, Sets.newConcurrentHashSet());
+            }
```
Comment (ZiheLiu) [inline fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: Is it possible that two threads try to call `addCounter` with the same `parentName`?

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: Maybe use `ConcurrentHashSet#computeIfAbsent`?
```java
Set<String> childNames = childCounterMap.computeIfAbsent(parentName, k -> Sets.newConcurrentHashSet());
```

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: I think it is impossible

Comment (kangkaisen) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: /**
 * It is accessed by two kinds of thread, one is to create this RuntimeProfile
 * , named 'query thread', the other is to call
 * {@link com.starrocks.common.proc.CurrentQueryInfoProvider}.
 */

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: Oh, maybe it is impossible.
Only coordinator calls it, and `Coordinate#updateProfile` has protected by `synchronized`.

Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java:127]: It is concurrent safe to do so, one of `ConcurrentHashMap::putIfAbsent` success and the others fail, with no side effect.

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.4

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**